tests/pytests/test_aggregate.py

```diff
@@ -995,16 +992,37 @@ def test_aggregate_timeout():
             pipeline = conn.pipeline(transaction=False)
     pipeline.execute()
 
-    # On coordinator, we currently get a single empty result on timeout,
-    # because all the shards timed out but the coordinator doesn't report it.
-    env.expect('FT.AGGREGATE', 'idx', '*',
-               'LOAD', '2', '@t1', '@__key',
-               'APPLY', '@t1 ^ @t1', 'AS', 't1exp',
-               'groupby', '2', '@t1', '@t1exp',
-                    'REDUCE', 'tolist', '1', '@__key', 'AS', 'keys',
-               'TIMEOUT', '1',
-        ).equal( ['Timeout limit was reached'] if not env.isCluster() else [1, ['t1', None, 't1exp', None, 'keys', []]])
+def aggregate_test(protocol=2):
+    if VALGRIND:
+        # You don't want to run this under valgrind, it will take forever
+        raise unittest.SkipTest("Skipping timeout test under valgrind")
+
+    if protocol == 2:
+        env = Env(moduleArgs='DEFAULT_DIALECT 2 ON_TIMEOUT FAIL')
+    elif protocol == 3:
+        env = Env(moduleArgs='DEFAULT_DIALECT 2 ON_TIMEOUT FAIL', protocol=3)
+    else:
+        env.skip()
```

Comment:
```suggestion
    if VALGRIND:
        # You don't want to run this under valgrind, it will take forever
        raise unittest.SkipTest("Skipping timeout test under valgrind")
    elif protocol not in [2, 3]:
        # Unsupported protocol
        raise unittest.SkipTest("Unsupported protocol")

    env = Env(moduleArgs='DEFAULT_DIALECT 2 ON_TIMEOUT FAIL', protocol=protocol)
```